### PR TITLE
sync_diff_inspector: Unified time zone

### PR DESCRIPTION
**Pull request by Leavrth**
### What problem does this PR solve? 

downstream don't initialize the specific time zone when the connection is created.

### What is changed and how it works?

use the same time_zone parameter when the upstream/downstream's connection is created.

### CODE REVIEW

File: sync_diff_inspector/utils/utils.go [hunk 1]
```diff
@@ -723,6 +726,7 @@ func ResetColumns(tableInfo *model.TableInfo, columns []string) *model.TableInfo
 	for i, col := range tableInfo.Columns {
 		col.Offset = i
 		colMap[col.Name.O] = i
+		hasTimeStampType = hasTimeStampType || (col.FieldType.Tp == mysql.TypeTimestamp)
```
Comment (3pointer) [inline sync_diff_inspector/utils/utils.go:729]: Should we handle TypeDatetime also?

Comment (Leavrth) [reply sync_diff_inspector/utils/utils.go:729]: The time zone does not affect datetime.

**Review by 3pointer — APPROVED** [review]

File: sync_diff_inspector/source/source.go [hunk 2]
```diff
@@ -115,16 +117,18 @@ func NewSources(ctx context.Context, cfg *config.Config) (downstream Source, ups
 	tableDiffs := make([]*common.TableDiff, 0, len(tablesToBeCheck))
 	for _, tables := range tablesToBeCheck {
 		for _, tableConfig := range tables {
+			new_Info, needUnifiedTimeZone := utils.ResetColumns(tableConfig.TargetTableInfo, tableConfig.IgnoreColumns)
```
Comment (lance6716) [inline sync_diff_inspector/source/source.go]:
```suggestion
			newInfo, needUnifiedTimeZone := utils.ResetColumns(tableConfig.TargetTableInfo, tableConfig.IgnoreColumns)
```

**Review by lance6716 — COMMENTED** [review]
rest lgtm

**Review by lance6716 — APPROVED** [review]

**Pull request merged**